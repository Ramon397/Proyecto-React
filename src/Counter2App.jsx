import PropTypes from 'prop-types';

export const Counter2App = ({value2})=> {
  return (
    <>
        <h1>Counter2App</h1>
        <h2>{value2}</h2>
    </>
  )
}

Counter2App.propTypes={
    value2:PropTypes.string
}