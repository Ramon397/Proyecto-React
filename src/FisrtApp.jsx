import PropTypes from 'prop-types';
const newMessage ='Ramon';
const Objeto={
    message:'Hola',
    title:'Baco'
}
const Salud=(saludo)=>{
    return (`Hola soy ${saludo}`);
}

export const FirstName =({
    title,
    subTitle,
    name2
}) =>{
    
    //console.log(props); 
    return(
//        <Fragment> el fragmaent se pude omitir solo con (<> </>) es igual sin la necesidad de la importacion
        <>
            <h2>{title}</h2>
            <h2>{subTitle}</h2>           
            <h2>{name2}</h2> 
            <h1>Nombre: {newMessage}</h1>
            <h3>{1+1}</h3>{/*impresion de expresion*/}
            <code>{JSON.stringify(Objeto)}</code>{/*impresion de objeto*/}
            <p>Soy un subtitulo</p>
            <h1>{Salud('Juan')}</h1>
        </>
//        </Fragment>
    );
}

FirstName.propTypes={
    title:PropTypes.string.isRequired,//poner requeridos los props  y del tipo requerido si no marca error
    subTitle:PropTypes.string.isRequired
}
FirstName.defaultProps={ //esto entran antes que los protypes
    name2:'Ramon Eduardo',
    title:'No hay titulo',
    subTitle:'No hay subtitulo'
   
}