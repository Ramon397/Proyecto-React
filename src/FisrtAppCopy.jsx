import PropTypes from 'prop-types';
export const FirstName2 =({
    title,
    subTitle,
    name2
}) =>{
    
    //console.log(props); 
    return(
//        <Fragment> el fragmaent se pude omitir solo con (<> </>) es igual sin la necesidad de la importacion
        <>
            <h2>{title}</h2>
            <h2>{subTitle}</h2>           
            <h2>{name2}</h2> 
            
        </>
//        </Fragment>
    );
}

FirstName2.propTypes={
    title:PropTypes.string.isRequired,//poner requeridos los props  y del tipo requerido si no marca error
    subTitle:PropTypes.string.isRequired
}
FirstName2.defaultProps={ //esto entran antes que los protypes
    name2:'Ramon Eduardo',
   // title:'No hay titulo',
    subTitle:'No hay subtitulo'
   
}