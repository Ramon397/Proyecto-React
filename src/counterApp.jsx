import { useState } from 'react';//para uso de hooks
import PropTypes from 'prop-types';

export const CounterApp = ({value}) => { 
  console.log('render');//cada vez que le das click a  un boton el componente se vuelve a ejecutar,y se mantiene el estado con el usaState
  const [counter,setCounter] = useState(value);
  const handleAdd=()=> {
    setCounter(counter+1);//agrega un nuemro al valor
    //setCounter((c)=>c+1); 
}
const handleSubtract=()=> setCounter(counter-1);//resta una 
const handleReset=()=> setCounter(value);//resetea el valor
  
  return (
    <>
        <h1>counterApp</h1>
        <h2>{counter}</h2>
        <button onClick={handleAdd}>+1</button>
        <button onClick={handleSubtract}>-1</button>
        <button onClick={handleReset}>reset</button>
    </>
  )
}


CounterApp.propTypes={
    value: PropTypes.number
}