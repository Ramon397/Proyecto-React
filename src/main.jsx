import React from 'react';
import ReactDOM from 'react-dom/client';
//import {App,HelloWorldApp} from './HelloWordApp'
//import { FirstName } from './FisrtApp';
import {FirstName2} from './FisrtAppCopy';
import './styles.css';
//import { CounterApp } from './counterApp';
//import { Counter2App } from './Counter2App';
//funcion con retorno
//function App(){
//    return <h1>Hola Mundo!!!</h1>;
//}
//renderizar
ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/*<App/>*/}
        {/*<HelloWorldApp/>*/}
        {/*<FirstName  />*/}
        < FirstName2 title='Hola, Soy Vegeta'/>
     {/*  <CounterApp value={10}/>*/}
     {/*  <Counter2App value2={"hola soy counter2"}/>*/}
    </React.StrictMode>
);
