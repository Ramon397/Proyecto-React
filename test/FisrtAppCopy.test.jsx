import { render } from '@testing-library/react';

import { FirstName2 } from '../src/FisrtAppCopy';

describe('Pruebas en FirstApp', () => {

    test('debe de hacer match con el snapshot', () => {     

        const title = 'Hola, soy Goku';
        const {container }= render ( <FirstName2 title={ title }/> );
        //console.log(container);
        expect(container).toMatchSnapshot();
     });

     test('Debe de mostrar el titulo en un h1', () => { 
         const title = 'Hola, soy Goku';
         const {container,getByText}= render ( <FirstName2 title={ title }/> );
         expect(getByText(title)).toBeTruthy();//que se encunetra

         const h1=container.querySelector('h1');
         expect(h1.innerHTML).toContain(title);
        
    });

 });