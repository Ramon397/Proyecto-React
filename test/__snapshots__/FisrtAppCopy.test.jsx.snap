// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp debe de hacer match con el snapshot 1`] = `
<div>
  <h2>
    Hola, soy Goku
  </h2>
  <h2>
    No hay subtitulo
  </h2>
  <h2>
    Ramon Eduardo
  </h2>
</div>
`;
